### DISCUSSION TITLE
Unexpected Behavior with Aliased Query Parameters in FastAPI

### DISCUSSION THREAD

**Discussion by harishthalluri** [fastapi/fastapi#13062] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated
from fastapi import FastAPI, Query

app = FastAPI()

@app.get("/employee/")
async def employee(
    emp_name : 
        Annotated[str | None,
                  Query(
                      min_length=5,
                      max_length=10,
                      title='name of the employee',
                      description='data from employee table',
                      alias='employee_name',
                      )]=None):
    if emp_name:
        return "employee name is {}".format(emp_name)
    else:
        return "Dont worry about it"
```

### Description

When I access the endpoint with http://127.0.0.1:8000/employee/?emp_name=apple, it unexpectedly enters the else block. However, when I use the alias, http://127.0.0.1:8000/employee/?employee_name=apple, the if block is executed as expected.
Is this the intended behavior? If so, could you please explain the rationale behind it? Or is there something I'm missing in my understanding of aliased query parameters?"

### Operating System

macOS

### FastAPI Version

0.115.6

### Pydantic Version

2.10.3

### Python Version

Python 3.11.4

**Comment by yvallois** [comment, marked as answer]
This is the behavior you find with pydantic. When you use alias, you expect the input data to use the alias (in this case username). However, when you use it, it's the name of the field you've defined that you use (name in my example).

```python
from pydantic import BaseModel, Field, ValidationError


class User(BaseModel):
    name: str = Field(alias="user_name")

if __name__ == "__main__":
    user = User.model_validate({"user_name": "John Doe"})
    print(user)
    # name='John Doe'
    try:
        user = User.model_validate({"name": "John Doe"})
    except ValidationError as e:
        print(e)
        # 1 validation error for User
        # user_name
        #   Field required [type=missing, input_value={'name': 'John Doe'}, input_type=dict]
        #     For further information visit https://errors.pydantic.dev/2.10/v/missing

```
FastAPi uses pydantic for validation, and Query is an object that inherits FieldInfo from pydantic, I think you have the expected behavior here.
